### PR TITLE
restores jasmine test framework

### PR DESCRIPTION
**Pull request by mdegani**
- bring in jasmine
- HTML coverage reporting
- set `noEmitHelpers:true` in tsconfig and used this polyfill https://github.com/ngParty/ts-helpers which makes istanbul coverage reports (though transpiled ES5) neat.
- update existing unit tests to jasmine

Connected to rangle/rangle-starter#93

### CODE REVIEW

**Comment by mdegani** [comment]
Source maps are not working properly for classes.

**Comment by SethDavenport** [comment]
LGTM. :shipit:

File: src/index.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import 'es6-promise';
 import '../shims/shims_for_IE';
 
 import 'angular2/bundles/angular2-polyfills';
+import 'ts-helpers';
```
Comment (SethDavenport) [inline src/index.ts]: Hey @mdegani is this needed in index.ts, or can we get away with just adding it to test.entry.ts?

File: src/reducers/counter.test.ts [hunk 2]
```diff
@@ -9,23 +9,23 @@ let state = counterReducer();
 describe('counter reducer', () => {
```
Comment (mdegani) [inline src/reducers/counter.test.ts]: I forgot to remove `import * assert from 'assert` (applies to all tests).

File: src/index.ts (hunk 1 above)
Comment (mdegani) [inline src/index.ts]: I found that it was required.  Without this loaded, you get an error `uncaught ReferenceError: __decorate is not defined`.

**Pull request merged**